### PR TITLE
MAINT: Remove dead code backporting py2.6 warnings

### PR DESCRIPTION
**Pull request by eric-wieser**
Since this is now in `np.testing._private`, it's no longer usable by the outside world anyway

### CODE REVIEW

File: numpy/testing/_private/utils.py [hunk 1]
```diff
@@ -1639,98 +1639,6 @@ def integer_repr(x):
         raise ValueError("Unsupported dtype %s" % x.dtype)
 
 
-# The following two classes are copied from python 2.6 warnings module (context
-# manager)
-class WarningMessage(object):
```
Comment (charris) [inline numpy/testing/_private/utils.py]: Need to import this from `warnings`.

Comment (charris) [reply numpy/testing/_private/utils.py]: Did that, waiting on tests.

**Comment by charris** [comment]
Should probably add a release note on the removals.

**Comment by eric-wieser** [comment]
@charris: A brief history:

* 1.13: Available at `np.testing.utils.WarningManager`
* 1.14: Moved to `np.testing.nose_tools.utils.WarningManager`
* 1.15-dev: Moved to `np.testing._private.utils`
* This patch: removed

It seems that the release note was due in 1.14, and that ship has sailed.

I don't have the full picture for what's going on with these moves - since I think you made them, can I ask you adjust the release notes however you see fit?

File: numpy/testing/_private/utils.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/testing/_private/utils.py]: Thanks, guess my search for uses missed that one

**Comment by charris** [comment]
The imports use `numpy.testing`,  but I see it only does `from ._private.utils import *`, so those classes were missing anyway. Imports from `numpy.testing.utils` are deprecated.

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**